lib/active_shipping/carriers/australia_post.rb

```diff
@@ -0,0 +1,249 @@
+require 'active_support/core_ext/object/to_query'
+
+module ActiveShipping
+  class AustraliaPost < Carrier
+    cattr_reader :name
+    @@name = 'Australia Post'
+
+    URL = 'https://digitalapi.auspost.com.au/'
+
+    PARCEL_ENDPOINTS = {
+      service: {
+        domestic:      'postage/parcel/domestic/service',
+        international: 'postage/parcel/international/service'
+      },
+      calculate: {
+        domestic:      'postage/parcel/domestic/calculate',
+        international: 'postage/parcel/international/calculate'
+      }
+    }.freeze
+
+    def requirements
+      [:api_key]
+    end
+
+    def find_rates(origin, destination, packages, options = {})
+      packages = Array(packages)
+
+      service_requests = packages.map do |package|
+        service_request = ServiceRequest.new(origin, destination, package, options)
+
+        service_request.parse(commit(service_request.url))
+        service_request
+      end
+
+      combined_response = CombinedResponse.new(origin, destination, packages, service_requests)
+
+      RateResponse.new(true, 'success', combined_response.params, combined_response.options)
+    end
+
+    def calculate_rates(origin, destination, packages, service_code, options = {})
+      packages = Array(packages)
+
+      calculate_requests = packages.map do |package|
+        calculate_request = CalculateRequest.new(origin, destination, package, service_code, options)
+
+        calculate_request.parse(commit(calculate_request.url))
+        calculate_request
+      end
+
+      combined_response = CombinedResponse.new(origin, destination, packages, calculate_requests)
+
+      RateResponse.new(true, 'success', combined_response.params, combined_response.options)
+    end
+
+    private
+
+    def commit(request_url)
+      ssl_get(request_url, headers)
+
+    rescue ActiveUtils::ResponseError, ActiveShipping::ResponseError => e
+      data = JSON.parse(e.response.body)
+
+      RateResponse.new(false, data['error']['errorMessage'], data)
```

Comment:
There is an assumption here that a parsed error body will always have a `error:` key. This can raise a `nil` error.